File: Assignment_4/Assignment4_4.py
"""-------------------------------------------------------------------------------------------
                Assignment4_4(Inbuilt Filter,Map and Reduce functions)
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
---------------------------------------------------------------------------------------------
Problem statement: This program implements Filter,Map and reduce functionality
                   Input : List of numbers acepted from user 
                   Filter : Number list should be even only
                   Map : Calculate the square of number from list
                   Reduce : Addition of all numbers from the list
------------------------------------------------------------------------------------------"""
from MarvellousFunctions import acceptElements
from functools import reduce
#------------------------------------------------------------------------------------------
# This function accepts the numbers from the user and applies 
# filter,Map and reduce functionality
#------------------------------------------------------------------------------------------
def main():
    numberList = acceptElements()
    print("---------------------------------------------------------------")
    evenNumbersList=list(filter((lambda num : (num%2==0)),numberList))     
    print("\nFiltered (Even number) list:",evenNumbersList)
    squareNumbersList=list(map(lambda num : (num**2),evenNumbersList))
    print("\nMapped data (Sqaure of a number) list:",squareNumbersList)
    additionNumbers = reduce((lambda A,B: A+B), squareNumbersList)
    print("\nReduced data (Addition of all numbers) :",additionNumbers)
    print("---------------------------------------------------------------")

#-----------------------------------------------------------------------------------
# This is main entry point of the program
#-----------------------------------------------------------------------------------
if __name__=="__main__":
    main()
#-----------------------------------------------------------------------------------

File: JavaPractice/Assignment_4/Assignment4_4.java
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collector;
import java.util.stream.Collectors;
import java.util.stream.Stream;

/*-------------------------------------------------------------------------------------------
                          Assignment4_4
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
---------------------------------------------------------------------------------------------
Problem statement: This program implements Filter,Map and reduce functionality
                   Input : List of numbers accepted from user 
                   Filter : Number list should be even only
                   Map : Calculate the square of number from list
                   Reduce : Addition of all numbers from the list
------------------------------------------------------------------------------------------*/

public class Assignment4_4{


	public static void main(String[] args) {

		ArrayList<Integer> numberList=Module.acceptNumbers();
		System.out.println("List Elements are:"+numberList);
		//all 3 Steps in 1
//				int result=numberList.stream()
//						.filter(a-> a%2==0)
//						.map(n->n*n)
//						.reduce(0,(a,b)->a+b);
//				System.out.println(result);


		
		// Filtered list of EVEN numbers only		
		List<Integer> evenNumberList=numberList.stream()
							.filter(a-> a%2==0)
							.collect(Collectors.toList());


		System.out.println("Filtered list of EVEN numbers only):"+evenNumberList);

		//Mapped data(Square numbers)
		List<Integer> squaredNums=evenNumberList.stream()
							.map(n->n*n)
							.collect(Collectors.toList());
		
		System.out.println("Mapped data(Square Numbers): "+squaredNums);
		
		//Reduced data(Addition of list numbers)
		int addNums=squaredNums.stream().reduce(0, (a,b)->a+b);
		System.out.println("Reduced data(Addition of list numbers) : "+addNums);
		 

	}
	

}
